src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -118,37 +123,844 @@ public DatabaseRecordActions(BlittableJsonTextWriter writer)
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.SqlConnectionStrings) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.SqlConnectionStrings));
+                            WriteSqlConnectionStrings(databaseRecord.SqlConnectionStrings);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.PeriodicBackups) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.PeriodicBackups));
+                            WritePeriodicBackups(databaseRecord.PeriodicBackups);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.ExternalReplications) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.ExternalReplications));
+                            WriteExternalReplications(databaseRecord.ExternalReplications);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.RavenEtls) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.RavenEtls));
+                            WriteRavenEtls(databaseRecord.RavenEtls);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.SqlEtls) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.SqlEtls));
+                            WriteSqlEtls(databaseRecord.SqlEtls);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.HubPullReplications) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.HubPullReplications));
+                            WriteHubPullReplications(databaseRecord.HubPullReplications);
+                        }
+
+                        if ((databaseRecordItemType & DatabaseRecordItemType.SinkPullReplications) != 0)
+                        {
+                            _writer.WriteComma();
+                            _writer.WritePropertyName(nameof(databaseRecord.SinkPullReplications));
+                            WriteSinkPullReplications(databaseRecord.SinkPullReplications);
+                        }
+
+                        break;
+                }
+            }
+
+            private void WriteHubPullReplications(List<PullReplicationDefinition> hubPullReplications)
```

Comment:
Can't we do something like `context.Write(_writer, pullReplication.ToJson())`, if possible would be much cleaner.